### PR TITLE
Add Code of Conduct

### PR DESCRIPTION
**Pull request by svbergerem**

### CODE REVIEW

**Comment by AugierLe42e** [comment]
> The use of sexualized language or imagery

O__O

**Comment by Realtin** [comment]
💜

**Comment by jaywink** [comment]
:+1:

I'm sure common sense will be applied and suddenly all our d* posts wont be monitored whether they contain NSFW stuff or not.. ;)

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery
+* Personal attacks
+* Trolling or insulting/derogatory comments
+* Public or private harassment
+* Publishing other's private information, such as physical or electronic addresses, without explicit permission
+* Other unethical or unprofessional conduct.
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct. By adopting this Code of Conduct, project maintainers commit themselves to fairly and consistently applying these principles to every aspect of managing this project. Project maintainers who do not follow or enforce the Code of Conduct may be permanently removed from the project team.
```
Comment (jaywink) [inline CODE_OF_CONDUCT.md]: I think this should mention who makes this decision, now it's a bit vague or not clear at all. A single member cannot decide to remove anyone. The removal could be just like addition, ie a vote where all the project core members (= github org) have one vote and majority decides.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Contributor Code of Conduct
+
+As contributors and maintainers of this project, and in the interest of fostering an open and welcoming community, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities.
+
+We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, or nationality.
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery
```
Comment (jaywink) [inline CODE_OF_CONDUCT.md]: I'd like this not to mean that one cannot post or reshare NSFW material, which is not imho a sign of disrespect to anyone. I'm not sure how to phrase that, so it should be maybe up to common sense to only react to cases where someone posts sexualized imagery that is somehow insulting.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (svbergerem) [inline CODE_OF_CONDUCT.md]: The CoC says maintainers “**may** be permanently removed”. This doesn't change the way how we add or remove team members. Of course, to remove someone from the team, you will still have to open a Loomio vote in the github organization group.

Comment (jaywink) [inline CODE_OF_CONDUCT.md]: OK yeah I guess that is fine.

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (svbergerem) [inline CODE_OF_CONDUCT.md]: > This code of conduct applies both within project spaces and **in public spaces when an individual is representing the project or its community**.

Comment (jaywink) [inline CODE_OF_CONDUCT.md]: Ah thanks, I somehow misread that, it's clear :+1:

**Comment by AugierLe42e** [comment]
I admit I'm a bit embarrassed by this code of conduct. I mean it says approxymately nothing more than what the common sense says and, in most of countries around the world, the law prohibits.

Though, it is extremely vague and opened to interpretations. Let's take the sentence : `Other unethical or unprofessional conduct`. I'd bet that more or less noone in this project could agree on a precise definition of what is unethical.

What disturbes me the most is the sentence: `The use of sexualized language or imagery`. I understand that you try by saying this to prevent sexual harrasment in particular against women and LGBT community and that all ok. But again: this really too vague and wide. As an exemple, *lolicon* (drawing reprensentation of young person and children in a sexy/sexual way) are prohibited in France but not n Japan.

The differences between cultures around the world might lead to big problems with this text and that could lead to heavy consequences for a project that tries to fight the cultural uniformisation imposed by Facebook and Google :/

**Comment by Realtin** [comment]
@AugierLe42e It may have not happen to the diaspora* project yet but every now and then some person could get involved with the project that makes everyone or only a few people uncomfortable and if that happens we have this document we can point to and explain why this persons behavior is not welcome here and if that person doesn't agree to stop behaving that way that they are also not welcome here. 

It is easier to have a document like that then just someone having to stand up and saying "hey **I** think you'r an asshole right now, please stop this". 

Basically we should discuss issues and pull requests in a professional manner. like you would do at work.
  
Like posting a award winning photograph of a warzone with butchered people on the ground is not prohibited by law in almost every country but would still be super inappropriate to post to a pull request here if you want to say that this pull request is dead.

**Comment by AugierLe42e** [comment]
@Realtin : I understand the goals an I absolutely approve them. But, like I was saying to @svbergerem on IRC, what wories me is absolutely not the goal, but what the documents in itself says taking in account the cultural differences all around the world. We, as european, will globally understand it the same way. I doubt this will be the same thing for an Japan or a Maxican. And this fear is reinforced by the fact that this is a very small group of european male (core-team) that will be in charge to enforece it.

I won't fight this but I just want to highlight point that make this document appear as a double-edged sword to me.

**Comment by jaywink** [comment]
@AugierLe42e as @svbergerem quoted:

> This code of conduct applies both within project spaces and in public spaces when an individual is representing the project or its community.

The key here is **representing**. Like here, it would be against this CoC for me to post a gif with bouncy boobies to this PR, but I wouldn't be restricted in doing so otherwise in a post for example. On the other hand, if I'm making a post about some diaspora development to social media, again, posting a gif with bouncy boobies would be inappropriate.

I totally agree we should add this CoC to the project, it's just a little detail saying to the world that anyone is welcome to contribute and we're not mean to them :)

**Comment by AugierLe42e** [comment]
> The key here is representing. 

So this also applies on IRC when we make jokes and during the fosdem when I can't have what I want as a background. on my computer.

**Comment by jaywink** [comment]
> So this also applies on IRC when we make jokes and during the fosdem when I can't have what I want as a background. on my computer.

Depends on whether you plan on making jokes that offend someone or have screen backgrounds that are offensive ;)

Augier, I'm pretty sure everyone can just use common sense. We're all pretty sensible people here.

**Comment by AugierLe42e** [comment]
> Depends on whether you plan on making jokes that offend someone or have screen backgrounds that are offensive ;)

Is a Mageia background ok? :D

Alright, let's give a try, then

**Comment by Realtin** [comment]
should we add an email address like coc@diasporafoundation.org, so people know where to report possible violations?

**Comment by jaywink** [comment]
> should we add an email address like coc@diasporafoundation.org, so people know where to report possible violations?

Good idea, but maybe team@diasporafoundation.org? Too many email aliases ;)

**Comment by Realtin** [comment]
@jaywink yep that's good!

**Comment by svbergerem** [comment]
> should we add an email address [...], so people know where to report possible violations?

Done.

**Comment by denschub** [comment]
Tagging this for 0.5.3.0 so we have it in our released codebase as soon as possible.

Just wondering if somebody is willing to prepare a blog post for our blog? I'm not sure if that's a blogable topic, but we haven't published a post since June.

**Comment by denschub** [comment]
Merged!
